exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -93,13 +93,15 @@ public void ensureAtLeastOneField(ComplexWriter writer) {
     if (!atLeastOneWrite) {
       // if we had no columns, create one empty one so we can return some data for count purposes.
       SchemaPath sp = columns.get(0);
-      PathSegment root = sp.getRootSegment();
+      PathSegment fieldPath = sp.getRootSegment();
       BaseWriter.MapWriter fieldWriter = writer.rootAsMap();
-      while (root.getChild() != null && !root.getChild().isArray()) {
-        fieldWriter = fieldWriter.map(root.getNameSegment().getPath());
-        root = root.getChild();
+      while (fieldPath.getChild() != null && ! fieldPath.getChild().isArray()) {
+        fieldWriter = fieldWriter.map(fieldPath.getNameSegment().getPath());
+        fieldPath = fieldPath.getChild();
+      }
+      if (fieldWriter.isEmptyMap()) {
```

Comment:
Another question:

> Basically change:
> 
> if (!atLeastOneWrite) {
> 
> to
> 
> if (fieldWriter.isEmptyMap()) {

Would that field writer be the writer from writer.rootAsMap(), or does the code still need the loop and its traversal?

(It seems that changing the "if (!atLeastOneWrite)" to (the simplified equivalent) of "if (true)" (and keeping the loop) work work, but I don't know what the minimum fix is (i.e., exactly how much of that loop and traversal is needed).)